### PR TITLE
e2e: fix runtime reproducibility test and also check `kata.contrast-node-installer-image`

### PR DESCRIPTION
**Pull request by miampf**
Follow up of #947. New successful run can be found [here](https://github.com/edgelesssys/contrast/actions/runs/11502616673).

### CODE REVIEW

File: .github/workflows/e2e_runtime-reproducibility.yml [hunk 1]
```diff
@@ -64,25 +71,34 @@ jobs:
...
-              for line in f:
-                  matrix, checksum = line.strip().split("=")
-                  if not checksum in seen:
-                      seen[checksum] = []
-                  seen[checksum].append(matrix)
-
+          for file in os.listdir("./checksums"):
+              checksum = pathlib.Path("./checksums", file).read_text()
+              if not checksum in seen:
+                  seen[checksum] = []
```
Comment (Freax13) [inline .github/workflows/e2e_runtime-reproducibility.yml:94]:
```suggestion
          seen = { pathlib.Path("./checksums", file).read_text() for file in os.listdir("./checksums") }
```

Comment (miampf) [reply .github/workflows/e2e_runtime-reproducibility.yml:94]: forgot to add info where checksum was seen, added it. List comprehensions like this aren't possible with the new change.

**Review by Freax13 — APPROVED** [review]

**Comment by katexochen** [comment]
@miampf could you separate the fix and the additional test into two distinct commits please (and rebase-merge later)?

**Comment by miampf** [comment]
> @miampf could you separate the fix and the additional test into two distinct commits please (and rebase-merge later)?

It will take some time until merge anyways since the test is not failing as expected. When it comes to merging it I will keep this in mind tho :)

**Comment by miampf** [comment]
The test _does_ fail if different values are detected (look [here](https://github.com/edgelesssys/contrast/actions/runs/11611379147), I just randomly generated strings instead of using a hash), however, it does not fail when comparing the images digests (look [here](https://github.com/edgelesssys/contrast/actions/runs/11577111422)), reason being that they are equal. @katexochen, since reproducibility should be broken but the digests are the same, should we compare something other than the image digest? If yes, what do you think would be a good option?

**Comment by katexochen** [comment]
> katexochen, since reproducibility should be broken but the digests are the same, should we compare something other than the image digest? If yes, what do you think would be a good option?

It's fine, the current reproducibility issues are really hard to reproduce, it would be to much work to detect that in a ci job. The test succeeds as this reproduces between the two machine, so there is no blocker to merge this from my perspective.

**Pull request merged**